opencensus/stats/exporters/stackdriver_exporter.py

```diff
@@ -0,0 +1,415 @@
+		the backend.
+		Optional. """
+		return self._bundle_delay_threshold
+
+	@property
+	def bundle_count_threshold(self):
+		""" bundle_count_threshold determines how many view data events
+		 can be buffered before batch uploading them to the backend.
+		 Optional. """
+		return self._bundle_count_threshold
+
+	@property
+	def resource(self):
+		""" Resource is an optional field that represents the Stackdriver
+		MonitoredResource, a resource that can be used for monitoring.
+		If no custom ResourceDescriptor is set, a default MonitoredResource
+		with type global and no resource labels will be used.
+		Optional. """
+		return self._resource
+
+	@property
+	def metric_prefix(self):
+		""" metric_prefix overrides the OpenCensus prefix of a stackdriver metric.
+		 Optional. """
+		return self._metric_prefix
+
+	@property
+	def default_trace_attributes(self):
+		""" default_trace_attributes will be appended to every span that is exported to
+		 Stackdriver Trace. """
+		return self._default_trace_attributes
+
+	@property
+	def default_monitoring_labels(self):
+		""" default_monitoring_labels are labels added to 
+		every metric created by this
+		exporter in Stackdriver Monitoring.
+
+		If unset, this defaults to a single label with key "opencensus_task" and
+		value "py-<pid>@<hostname>". This default ensures that the set of labels
+		together with the default Resource (global) are unique to this
+		process, as required by Stackdriver Monitoring.
+
+		If you set default_monitoring_labels, make sure that the Resource field
+		together with these labels is unique to the
+		current process. This is to ensure that there is only a single writer to
+		each TimeSeries in Stackdriver.
+
+		Set this to Labels to avoid getting the
+		default "opencensus_task" label. You should only do this if you know that
+		the Resource you set uniquely identifies this Python process. """
+		return self._default_monitoring_labels
+
+	def handle_error(err):
+		if self.on_error:
+			self.on_error(err)
+			print("Error exporting to Stackdriver: %s", err)
+			return
+
+class StackdriverStatsExporter(base.StatsExporter):
+	""" StackdriverStatsExporter exports stats to the Stackdriver Monitoring. """
+	def __init__(self, 
+				options = Options(), 
+				created_views = {}, 
+				client = None, 
+				default_labels = {}):
+		self._options = options
+		self._created_views = created_views # Views already created remotely
+		self._client = client
+		self._default_labels = default_labels
+
+	@property
+	def options(self):
+		return self._options
+
+	@property
+	def created_views(self):
+		return self._created_views
+
+	@property
+	def client(self):
+		return self._client
+
+	@property
+	def default_labels(self):
+		return self._default_labels
+
+	def set_default_labels(self, value):
+		self._default_labels = value
+
+	def on_register_view(self, view):
+		""" create metric descriptor for the registered view  """
+		if view:
+			err = self.create_measure(view)
```

Comment:
in python doing a `try except` clause is the way to go.
it isn't idiomatic to return errors such as it is in `Golang`

also, semi-related:
https://docs.quantifiedcode.com/python-anti-patterns/readability/asking_for_permission_instead_of_forgiveness_when_working_with_files.html